text/0000-cargo-wrapper.md

```diff
@@ -0,0 +1,170 @@
+$ neon new my-neon-lib
+     Created library `my-neon-lib` project
+$ cd my-neon-lib
+$ neon build
+    Updating registry `https://github.com/rust-lang/crates.io-index`
+ Downloading neon v0.1.20
+ Downloading neon-runtime v0.1.20
+ Downloading libc v0.2.33
+ Downloading neon-build v0.1.20
+   Compiling cslice v0.2.0
+   Compiling utf8-ranges v1.0.0
+   Compiling void v1.0.2
+   Compiling neon-build v0.1.20
+   Compiling gcc v0.3.54
+   Compiling libc v0.2.33
+   Compiling lazy_static v0.2.9
+   Compiling regex-syntax v0.4.1
+   Compiling unreachable v1.0.0
+   Compiling neon v0.1.20
+   Compiling my-neon-lib v0.1.0 (file:///home/dherman/Sources/my-neon-lib)
+   Compiling thread_local v0.3.4
+   Compiling memchr v1.0.2
+   Compiling aho-corasick v0.6.3
+   Compiling regex v0.2.2
+   Compiling neon-runtime v0.1.20
+    Finished release [optimized] target(s) in 37.4 secs
+$ neon run
+hello neon!
+```
+
+
+# Motivation
+[motivation]: #motivation
+
+The purpose of the `neon` CLI tool is to act like Cargo for Neon projects: a one-stop workflow tool for generating, building, and managing Neon projects. While abstract over some of the flags required to properly build a Neon project.
+
+## Modified project layout
+
+The current Neon project layout is set up so that the top-level directory is an npm package but not a Rust crate, and it _contains_ a Rust crate in the `native/` subdirectory. This means that different tools apply to different parts of the project (`npm` and `neon` to the project root; `cargo` to the `native/` subdirectory). This can also leave the user confused about which directory they should be in and which tool they should use at any given time.
+
+```
+my-neon-lib
+├── lib
+│   └── index.js
+├── native
+│   ├── build.rs
+│   ├── Cargo.toml
+│   ├── index.node
+│   └── src
+│       └── lib.rs
+├── package.json
+└── README.md
+```
+
+This RFC proposes collapsing this space of possibilities to **eliminate confusion and cognitive overhead** associated with knowing which directory a user must be in and which tool applies where. The new layout makes a Neon project **_both_ an npm package _and_ a Rust crate**. The `neon` tool _replaces_ the `cargo` executable entirely in a Neon programmer's workflow; that is, `cargo` should never be used in a Neon project. Moreover, the `neon` tool should work regardless of what subdirectory of the project a user happens to be in (just as cargo works today).
+
+```
+my-neon-lib
+├── addon.node
+├── build.rs
+├── Cargo.toml
+├── lib
+│   └── index.js
+├── package.json
+├── README.md
+└── src
+    └── lib.rs
+```
+
+## Redesigned `neon` CLI
+
+The long-term goal of the `neon` CLI should be to go away and be replaced by a pure `cargo` plugin, allowing developers to work entirely with a cargo-based workflow. However, this relies on cargo extension hooks that do not yet exist. So in the meantime, we will provide `neon` as a wrapper around `cargo`, and eventually deprecate this once we can move to `cargo` proper.
```

Comment:
Not sure if my comment here is relevant to this line exactly or not, but I'd like to be able to install via `cargo install neon`.